### PR TITLE
[DOCS] Add deprecation notice for system indices

### PR DESCRIPTION
**Pull request by lockewritesdocs**
In 7.16, directly accessing system indices is deprecated. If an application or API directly accesses system indices, we generate warnings in the deprecation log as indicated in #79633. PR #80095 will add a deprecation message for 7.16.

In 8.0, users _must_ add the `allow_restricted_indices` permission set to `true` on a user role to access
system indices. This PR adds a deprecation notice to the 8.0 migration docs so that we can inform users of this change and link to documentation for guidance.

**We should wait until #79679 is merged so that we can link directly to that documentation for more details.**

Preview link: https://elasticsearch_80028.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_indices_changes

Relates to #79679

### CODE REVIEW

**Review by williamrandolph — APPROVED** [review]
From the dev's perspective, I think this is all accurate, and will be complete once we can link to something that explains what system indices are. I can't speak to the product manager perspective.

File: docs/reference/migration/migrate_8_0/indices.asciidoc [hunk 1]
```diff
@@ -6,6 +6,24 @@
...
+Directly accessing system indices is deprecated, and may be prevented in a
+future version. If you must access a system index, create a security role with
+an index permission that targets the specific index and set the
+`allow_restricted_indices` permission to `true`. Refer to 
+{ref}/defining-roles.html#roles-indices-priv[indices privileges] for
+information on adding this permission to an index privilege.
+
+*Impact* +
+Accessing system indices directly results in warnings in the header of API
+responses. Use {kib} or the associated feature's {es} APIs to manage the data
```
Comment (bytebilly) [inline docs/reference/migration/migrate_8_0/indices.asciidoc:23]:
```suggestion
responses. If available, use {kib} or the associated feature's {es} APIs to manage the data
```

We may not have coverage for every feature, so we should mention to check availability.

**Review by bytebilly — CHANGES_REQUESTED** [review]
I left a suggestion, LGTM otherwise

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by lockewritesdocs** [comment]
After battling with a  gnarly rebase, I applied these changes in #83688.

**Pull request closed without merging**